lib/pure/os.nim

```diff
@@ -1716,15 +1716,52 @@ proc createSymlink*(src, dest: string) {.noWeirdTarget.} =
     if symlink(src, dest) != 0:
       raiseOSError(osLastError(), $(src, dest))
 
+when defined(windows) and not weirdTarget:
+  proc getFinalPathNameByHandleW(hFile: Handle, lpszFilePath: WideCStringObj,
+    cchFilePath: DWORD, dwFlags: DWORD
+  ): DWORD {.importc: "GetFinalPathNameByHandleW",
+    stdcall, dynlib: "Kernel32.dll".}
+
 proc expandSymlink*(symlinkPath: string): string {.noWeirdTarget.} =
   ## Returns a string representing the path to which the symbolic link points.
   ##
-  ## On Windows this is a noop, `symlinkPath` is simply returned.
   ##
   ## See also:
   ## * `createSymlink proc <#createSymlink,string,string>`_
+  ## * `symlinkExists proc <#symlinkExists,string>`_
   when defined(windows):
-    result = symlinkPath
+    if not symlinkExists(symlinkPath):
+      raise newException(OSError, symlinkPath & " is not symlink or does not exist")
```

Comment:
```suggestion
```

because it's redundant with the check a few lines below:
```nim
if handle == INVALID_HANDLE_VALUE:
  raiseOSError(osLastError(), symlinkPath)
```

unless i'm missing something